### ISSUE TITLE
Rules that express all constants are not allowed

### ISSUE THREAD

**Issue by mbudiu-vmw** [vmware/differential-datalog#97]
```
RHeapAllocation_Type(_heap, _type) :- var _heap = "<<main method array>>", var _type = "java.lang.String[]".
```

**Comment by mbudiu-vmw** [comment]
Is this solved?

**Comment by ryzhyk** [comment]
Sort of. We can now have rules that only have the head and not the body, e.g.,

```
Rel(5,true, {var x = true; x or false}).
```

But we do not yet allow rules that have the body, but the body only consists of constant expressions, e.g., this is _not_ allowed:

```
Rel(a,b,c) :- a = 5, b = true, c = false.
```

**Comment by mbudiu-vmw** [comment]
These two do not seem to be semantically equivalent; the first one does not have field names.
Otherwise we could write a pass to convert the second to the first.

**Comment by ryzhyk** [comment]
Yes, it should be easy to just compile the second form directly to DD rather than reducing it to the first form.

**Comment by ryzhyk** [comment]
Closed via #953
